src/org/zaproxy/zap/extension/api/ContextAPI.java

```diff
@@ -296,6 +302,23 @@ public ApiResponse handleApiView(String name, JSONObject params)
 			}
 			result = resultList;
 			break;
+		case VIEW_URLS:
+			resultList = new ApiResponseList(name);
+			Set<String> addedUrls = new HashSet<>();
+			for (SiteNode node : getContext(params).getNodesInContextFromSiteTree()) {
+				HistoryReference href = node.getHistoryReference();
+				if (HistoryReference.getTemporaryTypes().contains(href.getHistoryType())) {

```

Comment:
When we were talking the other day about passive scanning temp fuzzer messages you mentioned that alerts on them  weren't an issue. If a URL is added or represented only by an alert on a temp message would it be missed here?

Reply:
Yes. I added that check to keep the same behaviour as the GUI functionality, I'm fine changing both places to ignore the type? (Maybe it's better, we already do that in other endpoints, e.g. #4176).

Reply:
I think that’s probably best (change both).

Reply:
OK, will do.